sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonIndexerTest.java

```diff
@@ -0,0 +1,115 @@
+
+  @Before
+  public void init() throws IOException {
+    context = SensorContextTester.create(baseDir);
+    workDir = Files.createTempDirectory("workDir");
+    context.fileSystem().setWorkDir(workDir);
+  }
+
+  @Test
+  public void test_indexer() {
+    PythonIndexer pythonIndexer = new PythonIndexer();
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    pythonIndexer.buildOnce(context, Arrays.asList(file1, file2));
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("main")).hasSize(1);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("mod")).hasSize(1);
+    Symbol modAddSymbol = projectLevelSymbolTable.getSymbol("mod.add");
+    assertThat(modAddSymbol).isNotNull();
+    assertThat(modAddSymbol.is(Symbol.Kind.FUNCTION)).isTrue();
+  }
+
+  @Test
+  public void test_indexer_removed_file() {
+    PythonIndexer pythonIndexer = new PythonIndexer();
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    pythonIndexer.buildOnce(context, Arrays.asList(file1, file2));
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
+
+    pythonIndexer.removeFile(file2);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("main")).hasSize(1);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("mod")).isNull();
+    Symbol modAddSymbol = projectLevelSymbolTable.getSymbol("mod.add");
+    assertThat(modAddSymbol).isNull();
+  }
+
+  @Test
+  public void test_indexer_added_file() throws IOException {
+    PythonIndexer pythonIndexer = new PythonIndexer();
+    InputFile file1 = inputFile("main.py");
+    InputFile file2 = inputFile("mod.py");
+    pythonIndexer.buildOnce(context, Arrays.asList(file1, file2));
+    ProjectLevelSymbolTable projectLevelSymbolTable = pythonIndexer.projectLevelSymbolTable();
+
+    InputFile file3 = createInputFile("added.py");
+    pythonIndexer.addFile(file3);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("main")).hasSize(1);
+    assertThat(projectLevelSymbolTable.getSymbolsFromModule("added")).hasSize(1);
+    Symbol newFuncSymbol = projectLevelSymbolTable.getSymbol("added.new_func");
+    assertThat(newFuncSymbol).isNotNull();
+    assertThat(newFuncSymbol.is(Symbol.Kind.FUNCTION)).isTrue();
+  }
+
+  private InputFile inputFile(String name) {
+    DefaultInputFile inputFile = createInputFile(name);
+    context.fileSystem().add(inputFile);
+    return inputFile;
+  }
+
+  private DefaultInputFile createInputFile(String name) {
+    return TestInputFileBuilder.create("moduleKey", name)
+      .setModuleBaseDir(baseDir.toPath())
+      .setCharset(StandardCharsets.UTF_8)
+      .setType(InputFile.Type.MAIN)
+      .setLanguage(Python.KEY)
+      .initMetadata(TestUtils.fileContent(new File(baseDir, name), StandardCharsets.UTF_8))
+      .build();
+  }
```

Comment:
Maybe we can move those methods in TestUtils, as they are shared between this test and `PythonSensorTest`

Reply:
Indeed! I'd like to address this in my next PR as well as I will update both test classes there.